assets/js/src/subscribers/importExport/step_results.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import MailPoet from 'mailpoet';
+import _ from 'underscore';
+import ReactStringReplace from 'react-string-replace';
+
+function renderResultMessage(subscribersCount, segments, initialMessage) {
+  if (subscribersCount) {
+    let message = ReactStringReplace(initialMessage, '%1$s', () => (
+      <strong key="%1$s">{subscribersCount.toLocaleString()}</strong>
+    ));
+    message = ReactStringReplace(message, '%2$s', () => (
+      `"${segments.join('", "')}"`
+    ));
+    return (<p>{message}</p>);
+  }
+  return null;
+}
+
+function renderNoAction(createdSubscribers, updatedSubscribers) {
+  if (!createdSubscribers && !updatedSubscribers) {
+    return (<p>{MailPoet.I18n.t('importNoAction')}</p>);
+  }
+  return null;
+}
+
+function renderNoWelcomeEmail(addedToSegmentWithWelcomeNotification) {
+  if (addedToSegmentWithWelcomeNotification) {
+    return (<p>{MailPoet.I18n.t('importNoWelcomeEmail')}</p>);
+  }
+  return null;
+}
+
+function StepResults({
+  errors,
+  createdSubscribers,
+  updatedSubscribers,
+  segments,
+  addedToSegmentWithWelcomeNotification,
+  navigate,
+}) {
+  if (errors.length) {
+    MailPoet.Notice.error(_.flatten(errors));
+  }
+  return (
+    <>
+      <div className="updated">
+        {renderResultMessage(createdSubscribers, segments, MailPoet.I18n.t('subscribersCreated'))}
```

Comment:
I wonder why are you using simple functions instead of making them React components?

Reply:
No reason really. Those functions are specific to this results component so I left it here. Those could've been components, but I didn't see a reason so I left them this way.